mozilla_django_oidc_db/config.py

```diff
@@ -109,19 +109,26 @@ def store_config(request: HttpRequest) -> None:
     mozilla-django-oidc's callback view deletes the state key after it has validated it,
     so our :func:`lookup_config` cannot extract it from the session anymore.
     """
-    # The config_class key is added to the state in the OIDCInit.get method.
-    # TODO: verify that the state query param is present for error flows! Need to check
-    # the OAUTH2 spec for this, but according to ChatGeePeeTee if the request contains
-    # it, the callback must have it too.
-    state_key = request.GET.get("state")
-    if not state_key or state_key not in (
-        states := request.session.get("oidc_states", [])
-    ):
-        raise BadRequest("Could not look up the referenced config.")
-
-    state = states[state_key]
+    # Attempt to retrieve the config_class from the session, this only works for users
+    # that are actually logged in as Django users
+    if request.session.get("OIDCDB_CONFIG_CLASS", ""):
+        config_class = request.session.get("OIDCDB_CONFIG_CLASS", "")
+    else:
+        # The config_class key is added to the state in the OIDCInit.get method.
+        # TODO: verify that the state query param is present for error flows! Need to check
+        # the OAUTH2 spec for this, but according to ChatGeePeeTee if the request contains
+        # it, the callback must have it too.
+        state_key = request.GET.get("state")
+        if not state_key or state_key not in (
+            states := request.session.get("oidc_states", [])
+        ):
+            raise BadRequest("Could not look up the referenced config.")
+
+        state = states[state_key]
+        config_class = state.get("config_class", "")
```

Comment:
I think the precedence needs to be swapped - first check if we have a state parameter and look up the config there, and only if that doesn't yield anything, check if we have a config class in the session.

Consider the following scenario (and maybe a test is required for this?):

1. User logs in to admin with OIDC: init -> callback works as expected, config_class is set in session
2. User starts a form in Open Forms with DigiD OIDC -> init is okay, callback flow is entered and sees there is a config class session key. Now the admin config class is used for the DigiD login flow, and that is wrong.

Reply:
Do you think this test is more appropriate to put in this library, or in OF itself?

Reply:
discussed in office - a test in the library seems most appropriate

Reply:
For some reason I have difficulty making this test, I'm not sure if it should use the `keycloak_login` or if I should be using the callback and init stuff manually..